src/data_hub/_api.py

```diff
@@ -88,11 +89,20 @@ def _get_default_client(
             value["nucleus_api_route"],
         )
 
-        client = DeeporiginData(
-            token=access_token,
-            org_id=org_id,
-            base_url=base_url,
-        )
+        if use_async:
+            from deeporigin_data import AsyncDeeporiginData
+
+            client = AsyncDeeporiginData(
+                token=access_token,
+                org_id=org_id,
+                base_url=base_url,
+            )
+        else:
+            client = DeeporiginData(
+                token=access_token,
+                org_id=org_id,
+                base_url=base_url,
+            )
```

Comment:
i've implemented support for async, but we're not using it anywhere. in future we might